File: rust/src/models/price_input.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{billing_period_enum::BillingPeriodEnum, pricing::Pricing};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PriceInput {
    pub cadence: BillingPeriodEnum,

    pub currency: String,

    pub pricing: Pricing,
}

impl PriceInput {
    pub fn new(cadence: BillingPeriodEnum, currency: String, pricing: Pricing) -> Self {
        Self {
            cadence,
            currency,
            pricing,
        }
    }
}

File: typescript/src/models/priceInput.ts
// this file is @generated
import { type BillingPeriodEnum, BillingPeriodEnumSerializer } from "./billingPeriodEnum";
import { type Pricing, PricingSerializer } from "./pricing";

export interface PriceInput {
  cadence: BillingPeriodEnum;

  currency: string;

  pricing: Pricing;
}

export const PriceInputSerializer = {
  _fromJsonObject(object: any): PriceInput {
    return {
      cadence: BillingPeriodEnumSerializer._fromJsonObject(object["cadence"]),
      currency: object["currency"],
      pricing: PricingSerializer._fromJsonObject(object["pricing"]),
    };
  },

  _toJsonObject(self: PriceInput): any {
    return {
      cadence: BillingPeriodEnumSerializer._toJsonObject(self.cadence),
      currency: self.currency,
      pricing: PricingSerializer._toJsonObject(self.pricing),
    };
  },
};
